integration/teardown/teardown.go

```diff
@@ -0,0 +1,48 @@
+// +build k8srequired
+
+package teardown
+
+import (
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/client"
+)
+
+// Teardown e2e testing environment.
+func Teardown(c *client.AzureClientSet, g *framework.Guest, h *framework.Host) error {
+	var err error
+
+	{
+		h.DeleteGuestCluster("azure-operator", "azureconfig", "deleting host vnet peering: deleted")
+
+		// only do full teardown when not on CI
+		if os.Getenv("CIRCLECI") == "true" {
+			return nil
+		}
+	}
+
+	{
+		err = framework.HelmCmd("delete azure-operator --purge")
```

Comment:
Would be cool to move towards some more abstracted and parameterized version of this soon because we want to go with the helm client implementations anyway and relying on some helm command strings in the client side here will make the refactoring more exhausting. Just something to keep in mind.